src/org/mozilla/javascript/ScriptRuntime.java

```diff
@@ -825,8 +825,10 @@ public static BigInteger toBigInt(String s) {
     }
 
     public static Number toNumeric(Object val) {
-        if (val instanceof BigInteger) {
-            return (BigInteger) val;
+        // toNumber does not allow java.math.BigInteger.
+        // toNumeric allows java.math.BigInteger.
+        if (val instanceof Number) {
+            return (Number) val;
         }
```

Comment:
Should there be a javadoc description for the method that explains the difference you noted in your comments?

Reply:
ref. 1f3758b
Thanks! I wrote it.